verticapy/pipeline/_train.py

```diff
@@ -0,0 +1,222 @@
+    DecisionTreeRegressor,
+    DummyTreeClassifier,
+    DummyTreeRegressor,
+)
+from verticapy.machine_learning.vertica.tsa import ARIMA, ARMA, AR, MA
+
+from ._helper import execute_and_add
+
+SUPPORTED_FUNCTIONS = [
+    BisectingKMeans,
+    DBSCAN,
+    KMeans,
+    KPrototypes,
+    NearestCentroid,
+    MCA,
+    PCA,
+    SVD,
+    IsolationForest,
+    RandomForestClassifier,
+    RandomForestRegressor,
+    XGBClassifier,
+    XGBRegressor,
+    ElasticNet,
+    Lasso,
+    LinearRegression,
+    LogisticRegression,
+    PoissonRegressor,
+    Ridge,
+    BernoulliNB,
+    CategoricalNB,
+    GaussianNB,
+    MultinomialNB,
+    NaiveBayes,
+    KNeighborsClassifier,
+    KernelDensity,
+    KNeighborsRegressor,
+    LocalOutlierFactor,
+    LinearSVC,
+    LinearSVR,
+    DecisionTreeClassifier,
+    DecisionTreeRegressor,
+    DummyTreeClassifier,
+    DummyTreeRegressor,
+    ARIMA,
+    ARMA,
+    AR,
+    MA,
+]
+
+
+def training(
+    train: dict, vdf: vDataFrame, pipeline_name: str, cols: SQLColumns
+) -> Tuple[str, VerticaModel, str]:
+    """
+    Run the training step
+    of the pipeline.
+
+    Parameters
+    ----------
+    train: dict
+        YAML object which outlines the steps of the operation.
+    vdf: vDataFrame
+        The model trained in the training step.
+    pipeline_name: str
+        The prefix name of the intended pipeline to unify
+        the creatation of the objects.
+    cols: SQLColumns
+        ``list`` of the columns needed to deploy the model.
+
+    Returns
+    -------
+    return meta_sql, model, model_sql
+    str
+        The SQL to replicate the steps of the yaml file.
+    VerticaModel
+        The model created after training.
+    str
+        The SQL needed to retrain the model.
+    """
+    meta_sql = ""
+    if "train_test_split" in train:
+        info = train["train_test_split"]
+        test_size = 0.33 if "test_size" not in info else info["test_size"]
+        train_set, test_set = vdf.train_test_split(test_size)
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TRAIN_VIEW'} AS SELECT * FROM "
+            + train_set.current_relation()
+            + ";"
+        )
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TEST_VIEW'} AS SELECT * FROM "
+            + test_set.current_relation()
+            + ";"
+        )
+
+    else:
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TRAIN_VIEW'} AS SELECT * FROM "
+            + vdf.current_relation()
+            + ";"
+        )
+        meta_sql += execute_and_add(
+            f"CREATE OR REPLACE VIEW {pipeline_name + '_TEST_VIEW'} AS SELECT * FROM "
+            + vdf.current_relation()
+            + ";"
+        )
+
+    methods = list(train.keys())
+    methods = [method for method in methods if "method" in method]
+    for method in methods:
+        tf = train[method]
```

Comment:
What does tf stand for?

Reply:
It's my oversight, I reused it from transformation in _transform.py. But I think tm (training method) should be an apt abbreviation.